### PR TITLE
Fix Confidence and token timestamps for ASR 

### PR DESCRIPTION
**Pull request by BrandonWeng**
### Why is this change needed?
Title + also imporve return types, using the tdt hypothesis instead of just returning so many arrayss. 

For confidence, we were ahrd coding it, essentially useless

Token timestamp also was pretty much being hardcoded. We have the global timestamps based onthe frames now so we should just reuse that instead 

Added a --metadata flag to transcribe, it prints the start/end and conf
```bash
brandonweng@Brandons-MacBook-Pro FluidAudio % swift run fluidaudio transcribe GLM\ 4.5.wav --metadata
Building for debugging...
[1/1] Write swift-version--58304C5D6DBC2206.txt
Build of product 'fluidaudio' complete! (0.10s)
Audio Transcription
===================

Testing Audio Conversion
--------------------------
Original format:
  Sample rate: 48000.0 Hz
  Channels: 1
  Format: 1
  Duration: 296.16 seconds

StreamingAsrManager will automatically convert to 16kHz mono

Using batch mode with direct processing

Testing Batch Transcription
------------------------------
ASR Manager initialized successfully
Processing 296.16s of audio (4738558 samples)


==================================================
BATCH TRANSCRIPTION RESULTS
==================================================

Final transcription:
...

Metadata:
  Confidence: 1.000
  Duration: 296.160s
  Start time: 0.240s
  End time: 296.080s

Token Timings:
    [0] ' I' (id: 380, start: 0.240s, end: 0.480s, conf: 0.998)
    [1] ' think' (id: 4321, start: 0.480s, end: 0.640s, conf: 1.000)
    [2] ' we' (id: 750, start: 0.640s, end: 0.800s, conf: 1.000)
    [3] ' have' (id: 1647, start: 0.800s, end: 0.960s, conf: 1.000)
    [4] ' fin' (id: 1062, start: 0.960s, end: 1.200s, conf: 1.000)
    [5] 'ally' (id: 2274, start: 1.200s, end: 1.360s, conf: 1.000)
    [6] ' got' (id: 4580, start: 1.360s, end: 1.520s, conf: 1.000)
    [7] ' a' (id: 279, start: 1.520s, end: 1.760s, conf: 1.000)
    [8] ' real' (id: 3480, start: 1.760s, end: 1.920s, conf: 1.000)
    [9] ' comp' (id: 1469, start: 1.920s, end: 2.080s, conf: 1.000)
    [10] 'et' (id: 291, start: 2.080s, end: 2.320s, conf: 1.000)
    [11] 'itor' (id: 4397, start: 2.320s, end: 2.480s, conf: 1.000)
    [12] ' for' (id: 509, start: 2.480s, end: 2.720s, conf: 1.000)
    [13] ' ant' (id: 2339, start: 2.720s, end: 2.880s, conf: 0.957)
    [14] 'h' (id: 7882, start: 2.880s, end: 3.040s, conf: 1.000)
    [15] 'rop' (id: 2934, start: 3.040s, end: 3.200s, conf: 0.999)
    [16] 'ic' (id: 404, start: 3.200s, end: 3.360s, conf: 1.000)
```

the softmax calculation is quite efficient so no impact on the RTFx
```
================================================================================
FLEURS BENCHMARK SUMMARY
================================================================================

Language                  | WER%   | CER%   | RTFx    | Duration | Processed | Skipped
-----------------------------------------------------------------------------------------
English (US)              | 5.7    | 2.8    | 128.5   | 3442.9s  | 350       | -
French (France)           | 5.8    | 2.4    | 125.3   | 560.8s   | 52        | 298
German (Germany)          | 3.1    | 1.2    | 148.0   | 62.1s    | 5         | -
Italian (Italy)           | 4.3    | 2.0    | 148.0   | 743.3s   | 50        | -
Russian (Russia)          | 7.7    | 2.8    | 129.5   | 621.2s   | 50        | -
Spanish (Spain)           | 6.5    | 3.0    | 143.6   | 586.9s   | 50        | -
Ukrainian (Ukraine)       | 6.5    | 1.9    | 127.1   | 528.2s   | 50        | -
-----------------------------------------------------------------------------------------
AVERAGE                   | 5.6    | 2.3    | 135.7   | 6545.5s  | 607       | 298
```

### CODE REVIEW

File: Sources/FluidAudio/ASR/Streaming/StreamingAsrManager.swift [hunk 1]
```diff
@@ -308,7 +309,7 @@ public actor StreamingAsrManager {
             // Start frame offset is now handled by decoder's timeJump mechanism
 
             // Call AsrManager directly with deduplication
-            let (tokens, timestamps, _) = try await asrManager.transcribeStreamingChunk(
+            let (tokens, timestamps, confidences, _) = try await asrManager.transcribeStreamingChunk(
```
Comment (BrandonWeng) [inline Sources/FluidAudio/ASR/Streaming/StreamingAsrManager.swift:312]: Going to make more changes in the streaming PR so ignore these changes for now

**Comment by BrandonWeng** [comment]
> ## Overall Assessment
> This PR makes excellent improvements to ASR confidence calculation and token timestamp handling. The move from hardcoded values to actual model probabilities is a significant step forward for accuracy and reliability.
> 
> ### Key Strengths
> * ✅ **Real confidence scores**: Uses actual token probabilities from TDT decoder instead of placeholder values
> * ✅ **Proper token timestamps**: Implements global timestamp calculation based on frame indices
> * ✅ **Better API design**: Consolidates return values into `TdtHypothesis` struct for cleaner interfaces
> * ✅ **Enhanced tooling**: Added `--metadata` flag provides valuable debugging information
> * ✅ **Comprehensive updates**: All related components updated consistently (manager, decoder, tests)
> 
> ### Areas for Improvement
> **Performance Optimization Needed (Priority: High)** The new `softmax()` function allocates arrays for every token prediction, which could impact performance on long transcriptions. Consider:
> 
> * Pre-allocated reusable buffers
> * Fast path for single-element arrays
> * Caching when logits haven't changed
> 
> **Confidence Semantics (Priority: Medium)** The `calculateConfidence()` function now mixes token-level probability averaging with duration-based adjustments. This could confuse users about what the confidence represents. Consider documenting the confidence calculation methodology clearly.
> 
> **Memory Safety (Priority: Low)** TdtHypothesis helper methods don't validate array length consistency between tokens, timestamps, and confidences.
> 
> ### Recommendation
> **Approve after performance optimization**. The core improvements are valuable, but the softmax performance issue should be addressed before merging to production.

Tried the memory and performance optimizations , it actualy made performance worse. The arrays aren't that big so copy/ing deleting is probably taking more time

File: Sources/FluidAudio/ASR/MetalAcceleration.swift [hunk 2]
```diff
@@ -1,233 +0,0 @@
-import Accelerate
```
Comment (BrandonWeng) [inline Sources/FluidAudio/ASR/MetalAcceleration.swift:1]: not used

File: Sources/FluidAudio/ASR/MetalAcceleration.swift
Comment (Alex-Wengg) [inline Sources/FluidAudio/ASR/MetalAcceleration.swift:1]: why was this removed

Comment (BrandonWeng) [reply Sources/FluidAudio/ASR/MetalAcceleration.swift:1]: not needed, not used

File: Sources/FluidAudio/ASR/Streaming/StreamingAsrManager.swift [hunk 3]
```diff
@@ -328,6 +329,7 @@ public actor StreamingAsrManager {
             let interim = asrManager.processTranscriptionResult(
                 tokenIds: tokens,  // Only current chunk tokens for progress updates
                 timestamps: timestamps,
+                confidences: confidences,
```
Comment (Alex-Wengg) [inline Sources/FluidAudio/ASR/Streaming/StreamingAsrManager.swift:332]: will this let users decide the confident scores themselves

Comment (BrandonWeng) [reply Sources/FluidAudio/ASR/Streaming/StreamingAsrManager.swift:332]: Plan to use it in streaming - but we could expose a min confidence threshold to users as well

**Review by Alex-Wengg — APPROVED** [review]
